theano/sandbox/rng_mrg.py

```diff
@@ -1446,55 +1446,74 @@ def multinomial(self, size=None, n=1, pvals=None, ndim=None, dtype='int64',
-        # TODO : need description for parameter
+    def choice(self, size=None, a=2, replace=True, p=None, ndim=None,
+               dtype='int64', nstreams=None):
         """
-        Sample `n` times *WITHOUT replacement* from a multinomial distribution
-        defined by probabilities pvals, and returns the indices of the sampled
-        elements.
-        `n` needs to be in [1, m], where m is the number of elements to select
-        from, i.e. m == pvals.shape[1]. By default n = 1.
+        Sample `size` times from a multinomial distribution defined by
+        probabilities `p` and values between 0 and `a-1`.
 
-        Example : pvals = [[.98, .01, .01], [.01, .49, .50]] and n=1 will
-        probably result in [[0],[2]]. When setting n=2, this
+        Parameters
+        ----------
+        size: None
+            For now, size should be None. The number of samples is inferred
+            from the size of p.
+        a: integer scalar tensor
+            the values are generated between 0 and a-1.
+        replace: bool (default True)
+            Whether the sample is with or without replacement.
+            Only replace=False is implemented for now.
+        p: 2d numpy array or theano tensor
+            the probabilities of the distribution associated to each value of
+            `a`. It should have the same shape as a (if a is an array/tensor).
+
+        Example : p = [[.98, .01, .01], [.01, .49, .50]] and size=1 will
+        probably result in [[0],[2]]. When setting size=2, this
         will probably result in [[0,1],[2,1]].
 
         Notes
         -----
-        -`size` and `ndim` are only there keep the same signature as other
+        - `size` and `ndim` are only there keep the same signature as other
         uniform, binomial, normal, etc.
-        TODO : adapt multinomial to take that into account
 
-        -Does not do any value checking on pvals, i.e. there is no
+        - Does not do any value checking on pvals, i.e. there is no
         check that the elements are non-negative, less than 1, or
         sum to 1. passing pvals = [[-2., 2.]] will result in
         sampling [[0, 0]]
 
+        - Only replace=False is implemented for now.
+
         """
-        if pvals is None:
-            raise TypeError("You have to specify pvals")
-        pvals = as_tensor_variable(pvals)
+        if replace:
+            raise NotImplementedError(
+                "MRG_RandomStreams.choice only works without replacement "
+                "for now.")
+
+        if p is None:
+            raise TypeError("You have to specify p in "
+                            "MRG_RandomStreams.choice.")
+        p = as_tensor_variable(p)
 
-        if size is not None:
-            raise ValueError("Provided a size argument to "
-                             "MRG_RandomStreams.multinomial_wo_replacement, "
-                             "which does not use the size argument.")
```

Comment:
`size` is the number of samples that we want, it corresponds to the `n` of `multinomial_wo_replacement`. So I think that it should not be None (it should be an integer or integer tensor).